src/libraries/System.Data.Common/src/System/Data/DataColumn.cs

```diff
@@ -92,7 +98,9 @@ public DataColumn(string? columnName) : this(columnName, typeof(string), null, M
         /// Initializes a new instance of the <see cref='System.Data.DataColumn'/> class
         /// using the specified column name and data type.
         /// </summary>
-        public DataColumn(string? columnName, Type dataType) : this(columnName, dataType, null, MappingType.Element)
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:UnrecognizedReflectionPattern",
+            Justification = "Expression is null and `dataType` is marked appropriately.")]
+        public DataColumn(string? columnName, [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicProperties)] Type dataType) : this(columnName, dataType, null, MappingType.Element)
```

Comment:
It would be good to flow this DynamicallyAccessedMemberTypes attributes around to where it needs it internally. It is currently getting lost here since we are calling a RUC method, and suppressing the warning.